### ISSUE TITLE
Composer Lock File Warning when Going From Windows to Linux System

### ISSUE THREAD

**Issue by virgofx** [composer/composer#3044]
The following warning occurs when issuing a `composer install` on a Linux system where the Lock file was generated via a `composer update` on a Windows system.

```
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them.
Nothing to install or update
```

Windows system runs:
```
composer update
```

Resulting lock file contains:
```
{
   ...
    "hash": "f2fe4ec3805d2d172c3ff69e8ca1b22e",
    "packages": { ... },
    "packages-dev": { ... }
}
```

So far so good. Next, move to a Linux system and clone repository then attempt to install using `composer install`. The warning above happens. In order for the warning to be removed, one must first run `composer update`. When doing a compare the only difference in the lock file ends up being the value of the "hash". 

![image](https://cloud.githubusercontent.com/assets/739719/3237049/84e93890-f0e3-11e3-9fdc-4e787a54fe6d.png)

Logical conclusion -- the warning is based off the expected value of the hash; however, this is being calculated differently on windows/linux systems.

**Comment by sryze** [comment]
Couldn't composer just convert CRLF to LF before computing the hash?

**Comment by Seldaek** [comment]
Yes we could *just* but it still woudln't solve the issue entirely. If someone tackles it and we need to live with two different hashes in the lock file for a while, I would rather have it well thought through this time around than a quick fix.

**Comment by slbmeh** [comment]
I've seen an issue somewhere I can't seem to find to cross-link.  It discusses setting the checksum based on relevant data that would impact a different set of packages.

**Comment by Seldaek** [comment]
Fixed by #4140
